DataFormats/TrackReco/interface/TrackExtraBase.h

```diff
@@ -15,25 +16,29 @@
 
 #include "DataFormats/TrackingRecHit/interface/TrackingRecHitFwd.h"
 #include "DataFormats/TrajectoryState/interface/LocalTrajectoryParameters.h"
+
 namespace reco
 {
 
 class TrackExtraBase
 {
-
 public:
     using TrajParams = std::vector<LocalTrajectoryParameters>;
     using Chi2sFive = std::vector<unsigned char>;
 
     /// default constructor
-    TrackExtraBase() : m_firstHit(-1), m_nHits(0) { }
+    TrackExtraBase() : m_firstHit((unsigned int) -1), m_nHits(0) { }

```

Comment:
would it be better to initialise this to `0` instead ?

Reply:
It is evident that ```m_firstHit``` is never used at its initialization value: therefore either "0" or the number corresponding to the maximum size result in the same final behavior. Maybe, having 0 hits and the first one located in the maximum possible unsigned int number as initialization values can make sense, even though I agree with you that initializing an unsigned type to "-1" can look odd.
However, your proposed fix silences clang without modifying anything in what was originally there: I would stick on it.